main.go

```diff
@@ -6,16 +6,32 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"net/textproto"
 	"os"
+	"strings"
 )
 
+type headersMap textproto.MIMEHeader
+
+func (i *headersMap) String() string {
+	return ""
+}
+
+func (i *headersMap) Set(value string) error {
+	v := strings.Split(value, ": ")
```

Comment:
Last change request: use `strings.SplitN` instead, as `strings.Split` would fail to set the proper value in `Foo: Bar:Baz` (it would return a slice of `len` 3). I forgot about it in the previous review 😊